client/components/TargetedPublishing/RouteSelect.jsx

```diff
@@ -55,11 +65,36 @@ class RouteSelect extends Component {
               No routes defined
             </option>
           )}
-          {this.state.routes.map((route, index) => (
-            <option value={route.id} key={route.id}>
-              {route.name}
-            </option>
-          ))}
+
+          {collectionRoutes.length && (
+            <optgroup label="Collection">
```

Comment:
this is repeated, extract it to own function/component with label and array of routes as a props.